stock_picking_group_by_partner_by_carrier/report/report_delivery_slip.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2021 Camptocamp
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, models
+from odoo.tools import float_is_zero, float_round
+
+
+class DeliverySlipReport(models.AbstractModel):
+    _name = "report.stock.report_deliveryslip"
+    _description = "Delivery Slip Report"
+
+    @api.model
+    def get_remaining_to_deliver_data(self, report_lines):
+        """Return dictionaries encoding pending quantities to deliver
+
+        Returns a list of dictionaries, encoding the data to be displayed
+        at the end of the delivery slip, summarising for each order the
+        pending quantities for each of its lines.
+        """
+        remaining_to_deliver_data = []
+        for report_line in report_lines:
```

Comment:
There can be sales order lines that are not delivered at all and so do not make a report_lines. You will miss them in the loop here